src/modules/party/dto/post-party-request.dto.ts

```diff
@@ -0,0 +1,45 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsISO8601, IsOptional, Length } from 'class-validator';
+
+export type PostPartyRequest = PostPartyRequestItem[];
+
+export class PostPartyRequestItem {
+  @ApiProperty()
+  @Length(3, 20)
+  alternateIdentifier: string;
```

Comment:
Can we add description that this is UKEF id for party.
It should be 8 characters long.
Just digits needs to be allowed.
Example for alternativeIdentifier can be 00291013 or 00654321

Please add examples for all fields, to be easier to understand  and document request.

Reply:
Good shout, I've added validation to check that the `alternateIdentifier` is exactly 8 characters long and only consists of digits and examples for all fields. Some questions:

1. The description for `citizenshipClass` on Anypoint Exchange is '"If Customer domicile country is UK set to '1' otherwise '2'". Should we validate that `citizenshipClass` is either 1 or 2? At the moment the Mulesoft code does not check for this.
2. According to Anypoint Exchange `officerRiskDate` can be `null`, but in reality trying this out on Postman causes it to error. What should we do when `officerRiskDate` is `null`?

Reply:
1. Not sure how much we should limit input. We can discuss in standup.
2. In Mulesoft we used same content type definition for GET and POST. 
  - GET. This field can be null for some legacy data, so for GET calls we need to be careful when doing data manipulations.
  - POST. no one reported bug that officerRiskDate = null causes crash, so looks like field was always provided. We can make it as required field for POST.